### PR TITLE
Add unit testing for activity creation when cancelling a recurring, related cleanup

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Unit testing & cleanup around cancelling a recurring contribution 

Before
----------------------------------------
Less testing

After
----------------------------------------
More testing

Technical Details
----------------------------------------
This is preliminary cleanup on supporting the new cancel_reason field on the form

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/BAO/ContributionRecur.php [hunk 1]
```diff
@@ -261,11 +261,9 @@ public static function deleteRecurContribution($recurId) {
    * @param array $params
    *   Recur contribution params
    *
-   * @param array $activityParams
-   *
    * @return bool
    */
-  public static function cancelRecurContribution($params, $activityParams = []) {
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/ContributionRecur.php]: we now pass in 2 vars in params membership_id & processor_message

File: CRM/Contribute/BAO/ContributionRecur.php [hunk 2]
```diff
@@ -274,6 +272,11 @@ public static function cancelRecurContribution($params, $activityParams = []) {
     if (!$recurId) {
       return FALSE;
     }
+    $activityParams
```
Comment (seamuslee001) [inline CRM/Contribute/BAO/ContributionRecur.php]: Shouldn't these 2 lines be on one? @eileenmcnaughton

Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/ContributionRecur.php]: I thought jenkins was the style police :-)

Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/ContributionRecur.php]: @seamuslee001 fixed & passing

**Comment by seamuslee001** [comment]
The extraction looks sensible and it has a unit tests and jenkins is passing so good with me merging

**Comment by eileenmcnaughton** [comment]
thanks @seamuslee001

**Pull request merged**